File: 1296-Divide Array in Sets of K Consecutive Numbers.py
from collections import Counter
class Solution(object):
    def isPossibleDivide(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        freq = Counter(nums)
        
        for key in sorted(freq):
            if freq[key]:
                for m in range(key+k-1, key, -1):
                    if freq[m] < freq[key]:
                        return False
                    freq[m] -= freq[key]
                freq[key] = 0
            
        return True

File: JavaScript/1296-Divide Array in Sets of K Consecutive Numbers.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var isPossibleDivide = function(nums, k) {
    let freq = new Map();
    for (let num of nums){
        freq.set(num, (freq.get(num)||0) + 1 );
    }
    let keys = [...freq.keys()].sort((a,b)=>a-b);
    
    for (let key of keys){
        if (freq.get(key) > 0){
            for (let m = key + k -1; m > key; m--){
                if (freq.get(m) >= freq.get(key)){
                    freq.set(m, freq.get(m) - freq.get(key))
                }else{
                    return false;
                }
            }
            freq.set(key, 0);
        }
    }
    return true;
};